platforms/documentation/docs/src/docs/userguide/optimizing-builds/configuration_cache.adoc

```diff
@@ -536,6 +536,57 @@ And, when an incompatible task is scheduled to run, Gradle discards the configur
 
 Check the method documentation for more details.
 
+[[config_cache:integrity_check]]
+=== Using integrity checks to debug store and load issues
+By default, the configuration cache doesn't perform many integrity checks when reading the entry.
+This improves performance and reduces entry size.
+However, it makes it harder to pinpoint some store and load issues.
+For example, the incorrectly stored object may be discovered later, when reading data of some unrelated task.
+It may cause a type mismatch when restoring a value of the property or break some other invariant, resulting in a potentially misleading exception.
```

Comment:
💭 Not a native speaker, but here an opinionated suggestion on how to rephrase the paragraph. Feel free to ignore, pick parts, etc.

```text
Configuration cache avoids extra performance and on-disk space costs by performing minimal integrity checks as it writes and reads the data.
However, it can make it harder to troubleshoot cases when concurrency problems or misbehaving serializers interfere with the data.
For example, an incorrectly written object may be discovered only later when reading data of some unrelated task.
It may cause a type mismatch when restoring a value of the property or break some other invariant, resulting in a potentially misleading error messages. 
```